src/test/app/PayChan_test.cpp

```diff
@@ -1036,6 +1037,63 @@ struct PayChan_test : public beast::unit_test::suite
                     !rv[jss::result][jss::signature_verified].asBool());
             }
         }
+        {
+            // Try to explicitly specify secp256k1 and Ed25519 keys:
+            env (create (charlie, alice, channelFunds, settleDelay, charlie.pk()));
+            env.close();
+
+            auto const chan = to_string (channel (*env.current (), charlie, alice));
+
+            std::string cpk;
+            {
+                auto const r =
+                    env.rpc ("account_channels", charlie.human (), alice.human ());
+                BEAST_EXPECT (r[jss::result][jss::channels].size () == 1);
+                BEAST_EXPECT (r[jss::result][jss::channels][0u][jss::channel_id] == chan);
+                BEAST_EXPECT (r[jss::result][jss::validated]);
+                cpk = r[jss::result][jss::channels][0u][jss::public_key].asString();
+            }
+
+            // Try to authorize without specifying a key type, expect an error:
+            auto const rs =
+                env.rpc ("channel_authorize", "charlie", chan, "1000");
+            auto const sig = rs[jss::result][jss::signature].asString ();
+            BEAST_EXPECT (!sig.empty ());
+            {
+                auto const rv = env.rpc(
+                    "channel_verify", cpk, chan, "1000", sig);
+                BEAST_EXPECT(!rv[jss::result][jss::signature_verified].asBool());
+            }
+
+            // Try to authorize using an unknown key type, except an error:
+            auto const rs1 =
+                env.rpc ("channel_authorize", "charlie", "nyx", chan, "1000");
+            BEAST_EXPECT(rs1[jss::error] == "badKeyType");
+
+            // Try to authorize using secp256k1; the authorization _should_
+            // succeed but the verification should fail:
```

Comment:
That's unfortunate. One way to fix this would be to optionally provide the public key to `channel_authorize` and either:

1) check that public key can be correctly derived from the secret key for the inferred type 
2) If the type isn't given, and the public key is, infer the key type by trying to derive the public key from the secret key for the different types.

However, since we can't require that without breaking the API, I'm not sure how helpful that really is.

Reply:
I thought about/considered option (2) but it was too big a change. It would be possible to do by retrieving the channel and examining the public key, but I opted not to do that.

Generally, I’m allergic to commands like `channel_authorize` which require submitting things to the server. I’d rather deprecate it and provide a standalone tool to achieve this.